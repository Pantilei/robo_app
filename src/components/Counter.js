import React from "react";

class Counter extends React.Component {
  shouldComponentUpdate(prevProps, preState) {
    if (this.props.count !== prevProps.count) {
      return true;
    }
    return false;
  }
  render() {
    //console.log("counter");
    return (
      <div>
        <button id="counter" onClick={() => this.props.countFunc()}>
          Click me: {this.props.count}
        </button>
      </div>
    );
  }
}

export default Counter;
