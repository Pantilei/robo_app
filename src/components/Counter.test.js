import React from "react";
import { shallow } from "enzyme";
import Counter from "./Counter";

it("Testing Counter component", () => {
  let count = 0;
  const countFunc = function() {
    count++;
    return count;
  };

  const wrapper = shallow(<Counter count={count} countFunc={countFunc} />);
  //wrapper.find('[id="counter"]').simulate("click");
  //console.log(wrapper.props());
  expect(wrapper.props().count).toEqual(undefined);
});
