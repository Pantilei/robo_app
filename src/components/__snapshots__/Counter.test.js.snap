// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Counter component 1`] = `ShallowWrapper {}`;
